### ISSUE TITLE
Execute flag limited controllable parameters

### ISSUE THREAD

**Issue by aeslaughter** [idaholab/moose#15503]
## Reason

In some cases the ability to change/control parameter must be limited. For example, some simulations compute properties that must remain constant with time, but for performing stochastic simulations the parameter must be changed initially from what exists in the main input file.

## Design

When declaring a parameter as controllable an execution flag(s) can be provided. If this parameter is changed on a differing execution flag and error will occur.

```c++
declareControllable("foo", EXEC_INITIAL);
```

## Impact

This will be an extension to the existing API and add new functionality

**Comment by aeslaughter** [comment]
@bwspenc

**Comment by bwspenc** [comment]
Yeah, this sounds like a good way to get the behavior we would want.
